posthog/async_migrations/migrations/0004_replicated_schema.py

```diff
@@ -62,38 +48,6 @@
 """
 
 
-@dataclass(frozen=True)
-class TableMigrationData:
-    name: str
-    new_table_engine: MergeTreeEngine
-    kafka_table_names: List[str]
-    create_kafka_tables: List[str]
-
-    @property
-    def renamed_table_name(self):
-        return self.name
-
-    @property
-    def backup_table_name(self):
-        return f"{self.name}_backup_0004_replicated_schema"
-
-    @property
-    def tmp_table_name(self):
-        return f"{self.name}_tmp_0004_replicated_schema"
-
-
-@dataclass(frozen=True)
-class ShardedTableMigrationData(TableMigrationData):
-    rename_to: str
-    extra_tables: Dict[str, str]
-    materialized_view_names: List[str]
-    create_materialized_views: List[str]
-
-    @property
-    def renamed_table_name(self):
-        return self.rename_to
-
-
 class Migration(AsyncMigrationDefinition):
```

Comment:
We should remove the code all-together rather than have something that half-works but not really.

Reply:
We can't remove it all, because we need to keep something that blocks self-hosted users from upgrading 1.33 -> 1.39 etc, i.e. bypass this migration from ever being ran & that should say that async migration 0004 isn't done. We could change the code more, though new installs need to keep passing it.

Reply:
We can remove the is_required part. This would still fail on old installations if due to posthog_max_version

Reply:
I don't get this argument:
1. In the initial flow we always expect this to be returning True - there's never a reason not to, even if the underlaying code changes
2. If user updates beyond what's allowed in posthog_max_version we should be failing that with or without the is_required check.